File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        #Medthod 1
        """
        row=len(matrix)
        col=len(matrix[0])
        d=[]
        for i in range(row):
            for j in range(col):
                if matrix[i][j]==0:
                    d.append([i,j])
        for i in d:
            for r in range(row):
                matrix[r][i[1]]=0
            for c in range(col):
                matrix[i[0]][c]=0
                
        """
        #Method 2
        row=len(matrix)
        col=len(matrix[0])
        r=set()
        c=set()
        for i in range(row):
            for j in range(col):
                if matrix[i][j]==0:
                    r.add(i)
                    c.add(j)
        
        for i in r:
            for j in range(col):
                matrix[i][j]=0
        for i in range(row):
            for j in c:
                matrix[i][j]=0
                
            
            

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<vector<int>>temp;
        temp=matrix;
        int m=matrix.size();
        int n=matrix[0].size();
        cout<<m<<n<<endl;
        for(int i=0;i<m;++i){
            for(int j=0;j<n;++j){
                if(temp[i][j]==0){
                    int row=i;
                    int col=j;
                    while(row>=0){
                        matrix[row][j]=0;
                        row--;
                    }
                    row=i;
                    while(row<m){
                        matrix[row][j]=0;
                        row++;
                    }
                    while(col>=0){
                        matrix[i][col]=0;
                        col--;
                    }
                    col=j;
                    while(col<n){
                        matrix[i][col]=0;
                        col++;
                    }
                    
                }
            }
        }
        
    }
};